core/nuxeo-resource.js

```diff
@@ -310,10 +310,15 @@ import './nuxeo-connection.js';
         .execute(options)
         .then((response) =>
           response.text().then((text) => {
-            try {
-              return text ? JSON.parse(text) : {};
-            } catch (e) {
-              return { error: 'Invalid json' };
+            const contentType = response.headers ? response.headers.get('content-type') : options.headers.accept;
```

Comment:
I understand that the fallback to `options.headers.accept` is needed because because our mocked responses do not include the content-type header, and perhaps they should. I can live with this fallback for backwards compatibility, but I think we should create an improvement ticket to fix the mock, by adding `headers: new Map(Object.entries({'content-type': 'application/json'})),` to https://github.com/nuxeo/nuxeo-elements/blob/master/testing-helpers/nuxeo-mock-client.js#L73, wdyt @nmpcunha?

Reply:
👋🏼 
Yes, guess so. Probably they are already there and we are simply not include them. I believe we didn't had that scenario nor use case in mind. 🤔 
Would you be able to track task? Thanks!

Reply:
Anyway, contributions are welcomed too 😝 :trollface: